Add function to load PREM

santisoler opened the pull request:
Add function to load PREM data into `pandas.DataFrame`.
Add tests for the new function, entry in the documentation and a gallery example.
Add `pandas` as dependency.

Fixes #8 


**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [x] Add tests for new features or tests that would have caught the bug that you're fixing.
- [x] Add new public functions/methods/classes to `doc/api/index.rst`.
- [x] Write detailed docstrings for all functions/methods.
- [x] If adding new functionality, add an example to the docstring, gallery, and/or tutorials.

santisoler committed:
Add PREM to registry.txt

santisoler committed:
Add fetch_prem()

santisoler committed:
Improve fetch_prem docstring

santisoler committed:
Add fetch_prem tests

santisoler committed:
Add example for fetch_prem()

santisoler committed:
Add fetch_prem to api/index.rst

santisoler committed:
Add docstring for prem test functions

santisoler commented:
@leouieda 
I have a few questions:
* Should we add `pandas` as dependency?
* Once the documentation is build, the `pandas.DataFrame` has no link, how can we solve it?
* `pylint` raises an error with the `:math:\eta` string in the docstring of `fetch_prem()` function. I don't know why this would be an error.

leouieda commented on rockhound/prem.py:
```diff
@@ -0,0 +1,51 @@
+    Returns
+    -------
+    prem : :class:`pandas.DataFrame` or str
+        The loaded data or the file path ot the downloaded data.
```
Could you please add the names and units of the columns in the DataFrame?

leouieda commented on rockhound/prem.py:
```diff
@@ -0,0 +1,51 @@
+
+def fetch_prem(load=True):
+    """
+    Fetch the Preliminary Reference Earth Model (PREM) model.
```
```suggestion
    Fetch the Preliminary Reference Earth Model (PREM).
```

santisoler committed:
Reformat using `make format`

leouieda commented on rockhound/prem.py:
```diff
@@ -0,0 +1,51 @@
+
+
+def fetch_prem(load=True):
+    """
```
```suggestion
    r"""
```

You need to make this a raw string in order to have `\` in there. I think that's what pylint is complaining about.

leouieda reviewed (changes requested):
> Should we add pandas as dependency?

Yes please. Add it to `requirements.txt`, `setup.py`, `doc/install.rst` and `environment.yml`.

> Once the documentation is build, the pandas.DataFrame has no link, how can we solve it?

That's done by intersphinx. You'll need to add this line: https://github.com/fatiando/harmonica/blob/master/doc/conf.py#L34

leouieda committed:
Accepted suggestion from leouieda

Co-Authored-By: santis19 <santiago.r.soler@gmail.com>

leouieda committed:
Accepted suggestion from leouieda

Co-Authored-By: santis19 <santiago.r.soler@gmail.com>

santisoler committed:
Add columns names and units in prem

santisoler committed:
Add pandas as dependency

santisoler committed:
Edit prem example docstring to raw string

santisoler committed:
Improve styling on prem test functions

Pylint was complaining about how we were retrieving prem data as
attributes (e.g. `prem.radius`).
Changed by a dictionary-like style: `prem['radius']`.

santisoler commented:
@leouieda 
I've added `pandas` as dependency, fixed the `r"""` issue and added columns names a their units in `prem` `DataFrame`.

I've also fixed some issues regarding styling.
If you agree, this PR is ready to merge.

leouieda commented:
:+1: thanks, Santi!
